src/screens/Cart/index.tsx

```diff
@@ -0,0 +1,126 @@
+import React, {FC, useEffect, useState} from 'react';
+import AsyncStorage from '@react-native-async-storage/async-storage';
+import CartFull from './CartFull';
+import CartEmpty from './CartEmpty';
+import {API_URL, CART_TOKEN} from '../../constants';
+import Loader from '../../components/Loader';
+
+const Cart: FC = () => {
+  const [cartToken, setCartToken] = useState<string>('');
+  const [isLoading, setIsLoading] = useState(true);
+  const [cartData, setCartData] = useState(null);
+  const [productsList, setProductsList] = useState([]);
+
+  useEffect(() => {
+    // ToDo: move this logic to action
+    async function bootstrapAsync() {
+      try {
+        const token = await AsyncStorage.getItem(CART_TOKEN);
+        let parsedCartData = null;
+
+        if (token) {
+          setCartToken(token);
+
+          const response = await fetch(`${API_URL}/cart`, {
+            method: 'GET',
+            headers: {
+              'Content-Type': 'application/json',
+              'X-Spree-Order-Token': token,
+            },
+          });
+
+          parsedCartData = await response.json();
+
+          setCartData(parsedCartData);
+        }
+
+        const productIds =
+          parsedCartData?.data.relationships.line_items.data.map(({id}) => id);
+        const url = `${API_URL}/products?filter[ids]=${productIds.join()}`;
+
+        const response = await fetch(url, {
+          method: 'GET',
+          headers: {
+            'Content-Type': 'application/json',
+          },
+        });
+
+        const result = await response.json();
+        setProductsList(result.data);
```

Comment:
shouldn't it be moved under `token` check? If we don't have token, than we caanot make this request